### PR TITLE
cmd/tailscale/cli: make 'tailscale update' support Debian/Ubuntu apt

### PR DESCRIPTION
**Pull request by bradfitz**
Updates #6995

### CODE REVIEW

File: cmd/tailscale/cli/update.go [hunk 1]
```diff
@@ -222,13 +224,92 @@ func (up *updater) updateDebLike() error {
...
+		}
+		buf.Write(line)
+		buf.WriteByte('\n')
+	}
+	if hadCorrect || bytes.Equal(bytes.TrimSpace(was), bytes.TrimSpace(buf.Bytes())) {
+		// Unchanged or close enough.
+		return false, nil
+	}
+	if changes != 1 {
+		// No changes, but also no correct line. Add it.
```
Comment (maisem) [inline cmd/tailscale/cli/update.go:309]: `Add it` indicates you are trying to add it, but you aren't

File: cmd/tailscale/cli/update.go [hunk 2]
```diff
@@ -222,13 +224,92 @@ func (up *updater) updateDebLike() error {
...
+			})
+		}
+		buf.Write(line)
+		buf.WriteByte('\n')
+	}
+	if hadCorrect || bytes.Equal(bytes.TrimSpace(was), bytes.TrimSpace(buf.Bytes())) {
+		// Unchanged or close enough.
+		return false, nil
+	}
+	if changes != 1 {
```
Comment (maisem) [inline cmd/tailscale/cli/update.go:308]: can there be more than 1 change?

Comment (bradfitz) [reply cmd/tailscale/cli/update.go:308]: not in the happy path

but if they did something unexpected (this error path is unexpected stuff) then maybe

File: cmd/tailscale/cli/update.go (hunk 1 above)
Comment (bradfitz) [reply cmd/tailscale/cli/update.go:309]: whoops. deleted.

File: cmd/tailscale/cli/update.go (hunk 2 above)
Comment (bradfitz) [reply cmd/tailscale/cli/update.go:308]: clarified

**Review by maisem — APPROVED** [review]

**Pull request merged**